app.go

```diff
@@ -57,6 +57,8 @@ type App struct {
 	// An action to execute after any subcommands are run, but after the subcommand has finished
 	// It is run even if Action() panics
 	After AfterFunc
+	// Should the help message be hidden in case of error in the Before action
+	HideHelpOnBeforeError bool
```

Comment:
This seems too fine-grain for a top level API switch? I feel like there must be some other way to accomplish this without expanding the public API.

Reply:
Hello. I have feel the same while implementing this solution, however it's the only solution I've come up with to prevent behaviour change.

The other idea I have is to simply remove the help message, and let the user decide if he want to show help or not. But this change would break the old behaviour.

Reply:
We should discuss this in the issue here => https://github.com/urfave/cli/issues/1081, our understanding of this problem isn't mature enough to warrant a PR at this time.